.clang-format

```diff
@@ -0,0 +1,145 @@
+---
+Language: Cpp
+# BasedOnStyle:  LLVM
+AccessModifierOffset: -2
+AlignAfterOpenBracket: Align
+AlignArrayOfStructures: None
+AlignConsecutiveAssignments: Consecutive
+AlignConsecutiveBitFields: Consecutive
+AlignConsecutiveDeclarations: None
+AlignConsecutiveMacros: Consecutive
+AlignEscapedNewlines: Left
+AlignOperands: DontAlign
+AlignTrailingComments: true
+AllowAllArgumentsOnNextLine: false
+AllowAllParametersOfDeclarationOnNextLine: true
+AllowShortBlocksOnASingleLine: Empty
+AllowShortCaseLabelsOnASingleLine: false
+AllowShortFunctionsOnASingleLine: Empty
+AllowShortLambdasOnASingleLine: Empty
+AllowShortIfStatementsOnASingleLine: Never
+AllowShortLoopsOnASingleLine: false
+AlwaysBreakAfterDefinitionReturnType: None
+AlwaysBreakAfterReturnType: None
+AllowShortEnumsOnASingleLine: false
+AlwaysBreakBeforeMultilineStrings: true
+AlwaysBreakTemplateDeclarations: Yes
+BinPackArguments: false
+BinPackParameters: false
+BraceWrapping:
+  AfterCaseLabel: false
+  AfterClass: true
+  AfterControlStatement: Never
```

Comment:
Should definitely be `Always`. Braces attached after a control statement are rare in GoldenDict codebase.  As it is, there is **a lot** of diff like this:
```diff
-    for( size_t left = baseText.size(); left; )
-    {
-      if( *nextChar >= 0x10000 )
-      {
+    for ( size_t left = baseText.size(); left; ) {
+      if ( *nextChar >= 0x10000 ) {
```

Reply:
It is stupid, becase the code is using a lot of this style of if

```
if {
}
else {
}
```

The else is not the same line with `if`'s `}`

The old style is self-conflicting.

To ensure good looking of `else` this, the Control Statement must be never.

Reply:
I have encountered opening brace on the next line much more often than attached to the `if` line in GoldenDict's source code. `try`-`catch` is rare and doesn't matter much. Again, this is the style of almost all new code committed in the past 5 years.

Reply:
> there is a lot of diff like this:

What do u expect?? The code isn't written in a unified style. Of course, there are a lot. It is impossible to change no code nothing and have a working formatter.

Some style are 50%/50%. It doesn't matter, just choose anyone.

Reply:
> What's the conflict in the existing style

Why don't u try adjust the .clang-format by yourself? This codebase has two styles:

The bracket placements and if else/try catch depend on each other.

```
if (){
}
else {
}
```

and

```
if (){
} else {
}
```

If the change line after `if ()` then you get an insane shit:

```
if ()
{
}
else 
{
}
```

Reply:
> > The style is adjusted to minimize how much code will be changed if run clang-format over the entire code base.
> In this case your personal preference contradicts the stated principle of the pull request.

Then why are you citing "recent changes"??

Reply:
> If the change line after `if ()` then you get an insane shit:
> 
> ```
> if ()
> {
> }
> else 
> {
> }
> ```

That's exactly the most common style in GoldenDict source code by far. If you reject it as "insane", don't claim objectivity in the PR description and comments.

> Why don't u try adjust the .clang-format by yourself? 

That is certainly preferable to reading and replying to your belligerent and nonconstructive comments. Nothing good comes out of the review at this rate.

Reply:
Keep reviewing, this must be "always".

> Nothing good comes out of the review at this rate.

Are u retard or something? You are the reviewer, then review everything at once. GO.

Reply:
> Should definitely be Always

> That's exactly the most common style in GoldenDict source code by far

If you see some format stupid, don't say anything because this comes from your review.

Reply:
>  your belligerent and nonconstructive comments

You are using triple standards:

* Your one-liner
* "Recent" commits
* Not my style

What the fck do u want???

Reply:
![image](https://user-images.githubusercontent.com/20123683/235634278-b9179160-43e6-4766-a0de-4f168adc8c98.png)

Reply:
With such an attitude and language from the PR author, I won't be reviewing it, at least for a while. Your goal here appears to be insulting reviewers and wasting time anyway.

Reply:
Your attitude is the problem.

You are not going to merge anything because this is written in your DNA, in your personality, you cannot lead and cannot make a decision, thus you are afraid to press that merge button. You cannot even merge your own PR.

Without proving other people wrong, you cannot satisfy yourself by reviewing. 

>  I won't be reviewing it

I legitimately don't see any difference. You won't merge it anyway. After weeks of useless work to stratify your stupid reviews and nothing happens https://github.com/goldendict/goldendict/pull/1515

![image](https://user-images.githubusercontent.com/20123683/235634811-c530b1ab-f675-4746-b7f5-ed18544d9384.png)

🤮 🤮 🤮 🤮 🤮 🤮 🤮  You changed 4k lines in total, and you align yourself with another guy, shameless. 🤮 🤮 🤮 🤮 🤮 🤮 🤮

Reply:
Why r u so autistic that you don't even know who can be your friend?? https://github.com/goldendict/goldendict/issues/1424